docs/localizability.md

```diff
@@ -0,0 +1,152 @@
+# Localizability
+
+To enable localization, your strings must first be localizable.  Practically speaking, this means:
+
+* You put the string in a resource file (.resx).
+* You added necessary functional and contextual comments for the resource string.
+* You used the string resource instead of a string literal or format string literal.
+
+## Resource commenting
+
+[This internal site][0] has a good explanation of the what, why, and how of resource commenting.  In brief, resource comments can contain both:
+
+* functional comments, which enable tools to understand your intent
+* contextual comments, which enable human translators to understand your intent
+
+Both are important.
+
+For Arcade-based builds that use [OneLocBuild][1], adding resource comments can be done using the same Visual Studio Resource Editor that you use to add and modify resource strings in .resx files.
+
+Unfortunately, NuGet.Client does not use OneLocBuild, and any resource comments in its .resx files are [ignored][2].  The only way to comment resource strings is by editing .lci files in the NuGet.Build.Localization repository [here][3] using [lcxadmin.exe][4].  Having resource strings in one repository (NuGet.Client) and resource comments in another repository (NuGet.Build.Localization) is suboptimal.
+
+## How NuGet assemblies get localized (high-level)
+
+Although the steps below are for the dev branch, the process is similar for release branches.
+
+1. A PR merges into the dev branch.
+1. A NuGet.Client build generates a `localizationArtifacts` artifact.  This artifact contains newly built (but unsigned) assemblies and .lcg files.  The .lcg files contain information extracted from .resx files and are the primary input for later localization.
+1. The localization team grabs the `localizationArtifacts` artifact from a NuGet.Client build.
+1. The localization team localizes strings and merges a ["LEGO" PR][5] with localized strings (in .lcl files).
+1. During the next NuGet.Client build, the build adds the [NuGet.Build.Localization][6] repository as a submodule of the NuGet.Client repository and checks out a branch of the same name as the NuGet.Client repository (e.g.:  dev, release-6.7.x, etc.).  Localized resources are [made available][7] to localized builds through the `NuGetBuildLocalizationRepository` property.
+1. When the build completes, the product is localized using localizations base on an earlier commit.  (This means strings added since then won't be localized.)
+
+Typically, the data flow would look something like this.  However, because NuGet.Client's build ignores resource comments in .resx files, those comments and .lce files do not contribute to the localization process.
```

Comment:
Does this mean all the comments we've had in our resource strings have never been read by translators? I wonder if we have mis-translated strings, if so.

Reply:
Yes, and it's possible, though I don't have a good way of knowing.  I suppose user reports of mis-translated strings would be the first signal.  See [this comment](https://github.com/NuGet/NuGet.Client/pull/5408#discussion_r1319032128) too.

Ideally, we'd onboard to Arcade's OneLocBuild.  Then, we could add resource comments directly in our resource files ([like so](https://github.com/dotnet/sign/blob/ef0e6b3ef8281dff1d62cea34445bd88fc3e6714/src/Sign.Core/Resources.resx#L172)) and be confident that translators (machine or human) see them.

Both https://github.com/NuGet/Client.Engineering/issues/574 and https://github.com/NuGet/Client.Engineering/issues/1914 are related.